tests/integration/ceph_bucket_notification_test.go

```diff
@@ -122,38 +110,165 @@ func (s *ObjectSuite) TestBucketNotificationsInOrder() {
+		obc, err := k8sh.BucketClientset.ObjectbucketV1alpha1().ObjectBucketClaims(obcNamespace).Get(ctx, obcName, metav1.GetOptions{})
+		assert.Nil(t, err)
+		delete(obc.Labels, bucketNotificationLabelPrefix+notificationName)
+		_, err = k8sh.BucketClientset.ObjectbucketV1alpha1().ObjectBucketClaims(obcNamespace).Update(ctx, obc, metav1.UpdateOptions{})
+		assert.Nil(t, err)
+		// check whether existing bucket notification uneffected
+		var notificationPresent bool
+		for i := 0; i < 4; i++ {
+			notificationPresent = helper.BucketClient.CheckBucketNotifications(namespace, storeName, obcName, bucketname, notificationName, s3endpoint)
+			if !notificationPresent {
+				break
 			}
-			logger.Infof("%d bucket notifications found in: %v", len(notifications.TopicConfigurations), notifications)
-			for _, notification := range notifications.TopicConfigurations {
-				if *notification.Id == notificationName {
-					return true
-				}
-				logger.Infof("bucket notifications name mismatch %q != %q", *notification.Id, notificationName)
+			time.Sleep(5 * time.Second)
+		}
+		assert.False(t, notificationPresent)
+	})
+
+	t.Run("add new topic, notification to existing ObjectBucketClaim", func(t *testing.T) {
+		newNotificationName := "new-notification"
+		newTopicName := "new-topic"
+		err := helper.TopicClient.CreateTopic(newTopicName, storeName, httpEndpointService)
+		assert.Nil(t, err)
+		created := utils.Retry(12, 2*time.Second, "topic is created", func() bool {
+			return helper.TopicClient.CheckTopic(newTopicName)
+		})
+		assert.True(t, created)
+		err = helper.NotificationClient.CreateNotification(newNotificationName, newTopicName)
+		assert.Nil(t, err)
+		created = utils.Retry(12, 2*time.Second, "notification is created", func() bool {
+			return helper.NotificationClient.CheckNotification(newNotificationName)
+		})
+		assert.True(t, created)
+		obc, err := k8sh.BucketClientset.ObjectbucketV1alpha1().ObjectBucketClaims(obcNamespace).Get(ctx, obcName, metav1.GetOptions{})
+		assert.Nil(t, err)
+		obc.Labels[bucketNotificationLabelPrefix+newNotificationName] = newNotificationName
+		_, err = k8sh.BucketClientset.ObjectbucketV1alpha1().ObjectBucketClaims(obcNamespace).Update(ctx, obc, metav1.UpdateOptions{})
+		assert.Nil(t, err)
+		// check whether bucket notification added
+		notificationPresent := utils.Retry(12, 2*time.Second, "notification is created for bucket", func() bool {
+			return helper.BucketClient.CheckBucketNotifications(namespace, storeName, obcName, bucketname, newNotificationName, s3endpoint)
+		})
+		assert.True(t, notificationPresent)
+		err = helper.NotificationClient.DeleteNotification(newNotificationName, topicName)
+		assert.Nil(t, err)
+		err = helper.TopicClient.DeleteTopic(newTopicName, storeName, httpEndpointService)
+		assert.Nil(t, err)
+	})
+
+	t.Run("reverse order of creating notification,topic and adding it to ObjectBucketClaim", func(t *testing.T) {
```

Comment:
I think creating more subtest items will be particularly useful here.  Having the separate sections could help me read more closely at a glance what is happening and what can be expected based on that.